src/semant/translate.sml

```diff
@@ -181,12 +193,13 @@ structure Translate : TRANSLATE = struct
                             T.CONST(0))))
     end
 
-  fun for'(lo,hi,body,join) =
-    let val loR = T.TEMP(Temp.newtemp())
+  fun for'(loLoc,lo,hi,body,join) =
+    let val loR = followStaticLink(loLoc)
```

Comment:
Because it is simpler to just have this handle whether or not its in a register or frame. That being said, we still have to fix the followSL function to handle both cases